### PR TITLE
Add S3Event Parser for Lambda

### PR DESCRIPTION
**Pull request by danonorato**
1) Add S3EventParser class
2) Extend LambdaExecutor.java to call S3EventParser for s3 events
3) Add S3EventMappingTest.java
4) Add /resources/S3EventLambda.json

### CODE REVIEW

**Comment by danonorato** [comment]
@whummer  any idea what might be causing this test failure.  This PR doesn't get anywhere close to the Kinesis code base.  I made sure to merge in the latest upstream changes into this PR.  thoughts would be helpful.  regards,  Dan Onorato

**Comment by danonorato** [comment]
@whummer   checking in again on the above...  thx.

**Comment by danonorato** [comment]
@whummer   after further research it appears the several Java specs are broken in this repo.  The best that I can tell,  I don't see any PRs that have trigger the java tests in quite some time.  ie.. triggering ```$ make test-java-if-changed```  The BasicFunctionalityTest.testKinesisLambdaIntegration() spec will not pass as written and is suffer from split brain syndrome.  It uses LocalTestUtil.createFunctionCode(KinesisLambdaHandler.class)) to construct and set the Lambda code in the request.  This methods zips up the lambda.jar it creates.  Yet in lambda_api.py get_java_handler() has a check for java* run-time and zip and returns the validation exception causing the test failure.  The test also set java8 run-time.    What do you suggest for a path forward here?  I would really like to get the S3Event handling into the master branch.  regards, Dan

**Comment by danonorato** [comment]
@whummer   thx for the review.  I've been really busy with at work.  I'll get back to this PR soon to address your comments.  Regards,  Dan

**Comment by myrabetty** [comment]
Hello, is this still active? I would like to have this feature to test some code locally. I can totally help if needed.

**Comment by whummer** [comment]
@danonorato Any updates on this PR? Would be great if we could get this feature merged soon! Thanks

**Comment by whummer** [comment]
@myrabetty Do you think you could help out here and address the comments in a new PR that builds on top of this PR? (We want to make sure that the original commit from @danonorato is retained.) That would be great - very much appreciated!

**Comment by myrabetty** [comment]
Indeed. That is what I was thinking to do. I will likely have some time next week end.

File: localstack/ext/java/src/main/java/cloud/localstack/lambda/S3EventParser.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+import java.util.*;
+
+
+public class S3EventParser {
+
+    public static S3Event parse(List<Map<String, Object>> records) {
+
+        // parse out items to construct the S3EventNotification
+        Map<String, Object> record = records.get(0);
```
[the remark this answers is not in the document]
Comment (danonorato) [reply localstack/ext/java/src/main/java/cloud/localstack/lambda/S3EventParser.java]: S3 events come one at a time in this context.  @whummer  still slammed with current obligations.  I see you are trying to get this PR pushed along.  much appreciated.

**Comment by danonorato** [comment]
@whummer   I had a moment to address your review comments.  Please have a look.  Regards. Dan Onorato

**Comment by whummer** [comment]
Awesome - thanks a lot for updating the PR @danonorato !

**Pull request merged**